gdal/port/cpl_vsil.cpp

```diff
@@ -2362,79 +2361,66 @@ static VSIFileManager *poManager = nullptr;
 static CPLMutex* hVSIFileManagerMutex = nullptr;
 
 VSIFileManager *VSIFileManager::Get()
-
 {
     static volatile GPtrDiff_t nConstructerPID = 0;
-    if( poManager != nullptr )
+ 
     {
-        if( nConstructerPID != 0 )
-        {
-            GPtrDiff_t nCurrentPID = static_cast<GPtrDiff_t>(CPLGetPID());
-            if( nConstructerPID != nCurrentPID )
-            {
-                {
-                    CPLMutexHolder oHolder( &hVSIFileManagerMutex );
-                }
-                // This construct to avoid a lock is highly questionable
-                // and could probably fail with out-of-order CPU execution
-                // so the cppcheck warning is probably quite relevant.
-                // cppcheck-suppress identicalInnerCondition
-                if( nConstructerPID != 0 )
-                {
-                    VSIDebug1( "nConstructerPID != 0: %d", nConstructerPID);
-                    assert(false);
-                }
-            }
-        }
+      CPLMutexHolder oHolder(&hVSIFileManagerMutex);
+      if ( poManager != nullptr ) {
         return poManager;
+      }
     }
 
-    CPLMutexHolder oHolder2( &hVSIFileManagerMutex );
-    if( poManager == nullptr )
     {
-        nConstructerPID = static_cast<GPtrDiff_t>(CPLGetPID());
+      CPLMutexHolder oHolder( &hVSIFileManagerMutex );
+      nConstructerPID = static_cast<GPtrDiff_t>(CPLGetPID());
```

Comment:
all references to nConstructerPID should be removed

Reply:
Done.